R/1-5-0calculate_distance.R

```diff
@@ -1,23 +1,36 @@
 #' Calculate distance measure between original cutoffs and safe cutoffs
 #'
 #' @param org_cut Numeric vector of original cutoffs.
-#' @param safe_cut Data frame of safe cutoffs. The first column should be the group column, and subsequent columns the cutoffs.
-#'
-#' @return A numeric vector containing the L2 norms for each column of safe cutoffs.
+#' @param safe_cut Data frame of safe cutoffs. The first column should be the
+#'   group column, and subsequent columns the cutoffs.
+#' @return A numeric vector containing the distance between safe cutoffs and
+#'   original cutoffs, measured in terms of the L1 norm, L2 norm, and uniform
+#'   norm.
 #'
 #' @importFrom dplyr %>%
 #' @export
 calculate_distance <- function(org_cut, safe_cut) {
-  # Removing the group column
-  safe_cut_value <- safe_cut %>% select(-group)
-
-  # Function to calculate L2 norm
-  calculate_l2norm <- function(safe_col, org_cut) {
-    sqrt(sum((safe_col - org_cut)^2))
+  calculate_l1 <- function(safe_col, org_cut) {
+    sum(abs(safe_col - org_cut)) / length(org_cut)
+  }
+  calculate_l2 <- function(safe_col, org_cut) {
+    sum((safe_col - org_cut)^2) / length(org_cut)
```

Comment:
Technically the L2 norm has the `sqrt()`? 

Also, `sum(x) / length(x)` can be written as `mean(x)`.

Reply:
![Screenshot 2024-11-03 at 08 33 40](https://github.com/user-attachments/assets/91cfda7d-be60-4047-ae8d-76e285c671c7)